driver/docker/docker.go

```diff
@@ -385,3 +390,52 @@ func generateTar(files map[string]string) (io.Reader, error) {
 
 // ConfigurationOption is an option used to customize docker driver container and host config
 type ConfigurationOption func(*container.Config, *container.HostConfig) error
+
+// inspectImage inspects the operation image and returns an object of types.ImageInspect,
+// pulling the image if not found locally
+func (d *Driver) inspectImage(ctx context.Context, image bundle.InvocationImage) (types.ImageInspect, error) {
+	ii, _, err := d.dockerCli.Client().ImageInspectWithRaw(ctx, image.Image)
+	switch {
+	case client.IsErrNotFound(err):
+		fmt.Fprintf(d.dockerCli.Err(), "Unable to find image '%s' locally\n", image.Image)
+		if err := pullImage(ctx, d.dockerCli, image.Image); err != nil {
+			return ii, err
+		}
+		if ii, _, err = d.dockerCli.Client().ImageInspectWithRaw(ctx, image.Image); err != nil {
+			return ii, errors.Wrapf(err, "cannot inspect image %s", image.Image)
+		}
+	case err != nil:
+		return ii, errors.Wrapf(err, "cannot inspect image %s", image.Image)
+	}
+
+	return ii, nil
+}
+
+// validateImageDigest validates the operation image digest, if exists, against
+// the supplied repoDigests
+// TODO: or is it only acceptable for repoDigests to have one, exact match?
```

Comment:
It sounds like, indeed, it is only acceptable for repoDigests to have one, exact match.  Will update.